### PR TITLE
OSDOCS#6207: Installing a three-node Nutanix cluster

### PR DESCRIPTION
**Pull request by mjpytlak**
Version(s):
4.14+

Issue:
This PR addresses [osdocs-6278](https://issues.redhat.com/browse/OSDOCS-6278)

Link to docs preview:

- [Installing a three-node cluster on Nutanix](https://60612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_nutanix/installing-nutanix-three-node.html)
- [Creating the installation configuration file](https://60612--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_nutanix/installing-nutanix-installer-provisioned.html#installation-initializing_installing-nutanix-installer-provisioned) (note under step 2).

QE review:
- [X] QE has approved this change.

### CODE REVIEW

**Comment by mjpytlak** [comment]
@yanhua121 ready for SME review. PTAL. Thank you.

**Comment by yanhua121** [comment]
@mjpytlak 
The document for Nutanix 3-node IPI configuration looks good to me. Do we also need to mention the UPI for 3-node Nutanix cluster? Do we also need to test the UPI approach for 3-node Nutanix cluster, or the UPI is agnostic and does not need to test for individual provider?

**Comment by mjpytlak** [comment]
> Do we also need to mention the UPI for 3-node Nutanix cluster? Do we also need to test the UPI approach for 3-node Nutanix cluster, or the UPI is agnostic and does not need to test for individual provider?

@racedo I want to say that the agnostic UPI approach would be beyond the scope of this work, but can  you confirm?

**Comment by yanhua121** [comment]
> > Do we also need to mention the UPI for 3-node Nutanix cluster? Do we also need to test the UPI approach for 3-node Nutanix cluster, or the UPI is agnostic and does not need to test for individual provider?
> 
> @racedo I want to say that the agnostic UPI approach would be beyond the scope of this work, but can you confirm?

Yes, the agnostic UPI is out of the scope for this PR.

**Comment by mjpytlak** [comment]
> Do we also need to test the UPI approach for 3-node Nutanix cluster, or the UPI is agnostic and does not need to test for individual provider?

@sgaoshang Can you speak to this? Also - docs are ready for QE review. PTAL. Thank you.

**Comment by mjpytlak** [comment]
> Do we also need to mention the UPI for 3-node Nutanix cluster?

@yanhua121 I did not mention UPI for 3-node Nutanix cluster because we do not support a Nutanix specific UPI install.

**Comment by yanhua121** [comment]
> > Do we also need to mention the UPI for 3-node Nutanix cluster?
> 
> @yanhua121 I did not mention UPI for 3-node Nutanix cluster because we do not support a Nutanix specific UPI install.

@mjpytlak Okay, thanks. I think the PR looks good to me.

File: installing/installing_nutanix/installing-nutanix-three-node.adoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+include::_attributes/common-attributes.adoc[]
+:context: installing-nutanix-three-node
+
+toc::[]
+
+In {product-title} version {product-version}, you can install a three-node cluster on Nutanix. A three-node cluster consists of three control plane machines, which also act as compute machines. This type of cluster provides a smaller, more resource efficient cluster, for cluster administrators and developers to use for testing, development, and production.
+
+include::modules/installation-three-node-cluster-cloud-provider.adoc[leveloffset=+1]
+
+== Next steps
```
Comment (dfitzmau) [inline installing/installing_nutanix/installing-nutanix-three-node.adoc:13]: The next steps link might be confusing in that one has created a three-node cluster on Nutanix and then needs to create a generic cluster? Should the next steps addition be padded out to explain the next steps and then follow with a link or drop a user to the exact section in installing a cluster on Nutanix where they should continue?

Or maybe `Installing a three-node cluster on Nutanix` (renamed to configuring a three-node cluster environment on Nutanix) should be moved above `Installing a cluster on Nutanix`

File: modules/installation-initializing.adoc [hunk 2]
```diff
@@ -673,6 +673,11 @@ ifdef::nutanix[]
 . Optional: Update one or more of the default configuration parameters in the `install.config.yaml` file to customize the installation.
 +
 For more information about the parameters, see "Installation configuration parameters".
++
+[NOTE]
+====
+If you are installing a three-node cluster, be sure to set the `compute.replicas` parameter to `0`. This ensures that cluster's control planes are schedulable. For more information, see "Installing a three-node cluster on {platform}".
```
Comment (dfitzmau) [inline modules/installation-initializing.adoc:679]:
```suggestion
If you are installing a three-node cluster, be sure to set the `compute.replicas` parameter to `0`. This ensures that the cluster's control planes are schedulable. For more information, see "Installing a three-node cluster on {platform}".
```

File: modules/installation-three-node-cluster-cloud-provider.adoc [hunk 3]
```diff
@@ -38,16 +42,26 @@ Because application workloads run from control plane nodes, additional subscript
...
+apiVersion: v1
+baseDomain: example.com
 compute:
 - name: worker
   platform: {}
   replicas: 0
+# ...
 ----
-ifndef::vsphere,vmc[]
+ifndef::vsphere,vmc,nutanix[]
```
Comment (dfitzmau) [inline modules/installation-three-node-cluster-cloud-provider.adoc:64]: You might get a merge conflict here if https://github.com/openshift/openshift-docs/pull/60614 gets merged before your PR.

**Comment by dfitzmau** [comment]
Hi @mjpytlak . Nice work on this PR. I left some minor comments inline. I'll need @mburke5678 to do a scan of my review comments also.

File: installing/installing_nutanix/installing-nutanix-three-node.adoc (hunk 1 above)
Comment (mjpytlak) [reply installing/installing_nutanix/installing-nutanix-three-node.adoc:13]: The user has, in fact, not created a three-node cluster at this point. The purpose of this topic is to detail only those steps necessary to create a three node cluster. The purpose of the placement of this content in the TOC is for findability. If I am a customer looking to install a three-node cluster, Stephanie and I did not want to bury the information inside of an assembly. I am comfortable leaving the content as is, as this is the approach we took in 4.13, when introducing 3-node support for Azure, AWS, GCP, and vSphere.

File: modules/installation-initializing.adoc (hunk 2 above)
Comment (mjpytlak) [reply modules/installation-initializing.adoc:679]: Updated. Thank you.

File: modules/installation-three-node-cluster-cloud-provider.adoc (hunk 3 above)
Comment (mjpytlak) [reply modules/installation-three-node-cluster-cloud-provider.adoc:64]: Yeah. Probably, but easy enough to fix. Thanks.

File: installing/installing_nutanix/installing-nutanix-three-node.adoc (hunk 1 above)
Comment (mburke5678) [reply installing/installing_nutanix/installing-nutanix-three-node.adoc:13]: My concern is that this isn't truly a procedure that does a full action. It is more of an example. But, with the same format in 5 places, horse and barn, I reckon

File: modules/installation-three-node-cluster-cloud-provider.adoc [hunk 4]
```diff
@@ -38,16 +42,26 @@ Because application workloads run from control plane nodes, additional subscript
...
+.Example `install-config.yaml` file for a three-node cluster
 [source,yaml]
 ----
+apiVersion: v1
+baseDomain: example.com
 compute:
 - name: worker
   platform: {}
   replicas: 0
+# ...
```
Comment (mburke5678) [inline modules/installation-three-node-cluster-cloud-provider.adoc:62]: Is it possible to get the example to tuck under the numbered step as with the previous example?

**Comment by mburke5678** [comment]
@mjpytlak A couple of nits. Otherwise LGTM.

cc @dfitzmau

File: modules/installation-three-node-cluster-cloud-provider.adoc (hunk 4 above)
Comment (mburke5678) [reply modules/installation-three-node-cluster-cloud-provider.adoc:62]: image

Comment (mjpytlak) [reply modules/installation-three-node-cluster-cloud-provider.adoc:62]: That second example is for a UPI installation. A Nutanix specific UPI installation is not supported.

Comment (mburke5678) [reply modules/installation-three-node-cluster-cloud-provider.adoc:62]: I was just wondering if you could line it up under the numbered step. Not a big deal....

Comment (mjpytlak) [reply modules/installation-three-node-cluster-cloud-provider.adoc:62]: Oh. Sorry for the confusion @mburke5678. Was clearly overthinking your comment.

**Pull request merged**